client/core/core.go

```diff
@@ -913,12 +913,36 @@ func (c *Core) Run(ctx context.Context) {
 		c.latencyQ.Run(ctx)
 	}()
 
-	c.wg.Wait()
+	c.wg.Wait() // block here until all goroutines except DB complete
 
 	// Stop the DB after dexConnections and other goroutines are done.
 	stopDB()
 	dbWG.Wait()
 
+	// Lock and disconnect the wallets.
+	c.walletMtx.Lock()
+	for assetID, wallet := range c.wallets {
+		symb := strings.ToUpper(unbip(assetID))
+		c.log.Infof("Locking %s wallet", symb)
+		if err := wallet.Lock(); err != nil {
+			c.log.Errorf("Failed to lock %v wallet: %v", symb, err)
+		}
+		wallet.Disconnect()
```

Comment:
Have to consider cases like ReconfigureWallet where we don't want to lock before disconnect.  The sequence there is connect to new wallet, then disconnect from old, but the old one is very often the exact same wallet, just some different settings that are unrelated to RPC connection such as redeem conf target. So in that case you'd be locking the same wallet shared by both of the xcWallets.

Reply:
This was a good suggestion, but I decided to manually lock the wallet before disconnect in `CreateWallet` and `setWalletPassword` instead of making it part of the `Disconnect` method as it wasn't universally correct to lock before disconnect.  See https://github.com/decred/dcrdex/pull/928/commits/448be802856310c95ab3822e3e6c83353f2cbd45 just pushed.